core/src/main/java/org/jboss/galleon/layout/SystemPaths.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2016-2022 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.galleon.layout;
+
+import java.nio.file.Path;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class SystemPaths {
+
+    private final Set<Path> paths;
+
+    public SystemPaths(Set<Path> paths) {
```

Comment:
Are these supposed to be always relative or always absolute? It'be good to make sure they always are one or the other.

Reply:
Those should be relative paths only.
I added a check in `SystemPaths` and in `FeaturePackXmlParser30`